exchange/utils.go

```diff
@@ -43,6 +43,7 @@ func cleanOpenRTBRequests(ctx context.Context,
 
 	gdpr := extractGDPR(orig, usersyncIfAmbiguous)
 	consent := extractConsent(orig)
+	isAMP := (labels.RType == pbsmetrics.ReqTypeAMP) && gDPR.AMPException()
```

Comment:
Correct me if I am wrong here but `AMPException` flag being set means that we don't want AMP requests to be considered for GDPR and so keeping it `false` by default means that AMP requests will be subject to GDPR as well, right?

So based on that the above line will actually set `isAMP` to true only if the request is of type AMP and if the flag is set (meaning that we wanna exempt AMP requests from GDPR). Isn't that the opposite of what we want? Meaning that we want `isAMP` to be true only if the request is of type AMP and if the flag is not set i.e. we don't want AMP to be exempted from enforcing GDPR. In other words:
```
isAMP := (labels.RType == pbsmetrics.ReqTypeAMP) && !gDPR.AMPException()
```

Reply:
I read it as:
- `AMPException = true` Never scrub the buyer id or user id for AMP requests.
- `AMPException = false` Scrub the buyer id or user id for AMP requests per normal GDPR logic.

.. with the default of `false` not reverting the change back. So, the code as @hhhjort has it looks fine to me.